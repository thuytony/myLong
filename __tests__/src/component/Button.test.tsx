import 'react-native';
import React from 'react';
import {Text} from 'react-native';
import { Button } from '@components';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const instance = renderer.create(<Button />);
  expect(instance.toJSON()).toMatchSnapshot();
});

it('renders with title', () => {
  const instance = renderer.create(<Button title="test"/>);
  const textView = instance.root.findByType(Text);
  expect(textView.props.children).toBe('test');
});

it('renders with title all call', () => {
  const instance = renderer.create(<Button title="test" textAllCap={true}/>);
  const textView = instance.root.findByType(Text);
  expect(textView.props.children).toBe('TEST');
});
