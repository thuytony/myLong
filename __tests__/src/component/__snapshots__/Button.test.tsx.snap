// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#E3F8F4",
      "borderColor": "#99E1E1",
      "borderRadius": 5,
      "borderWidth": 2,
      "flexDirection": "row",
      "justifyContent": "center",
      "opacity": 1,
      "padding": 14,
    }
  }
/>
`;
